Kernel/FileSystem/Initializer.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2022, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Error.h>
+#include <AK/Span.h>
+#include <AK/StringView.h>
+#include <AK/Types.h>
+#include <Kernel/FileSystem/FileSystem.h>
+#include <Kernel/FileSystem/OpenFileDescription.h>
+
+namespace Kernel {
+
+struct FileSystemInitializer {
+    StringView short_name;
+    StringView name;
+    bool requires_open_file_description { false };
+    bool requires_block_device { false };
+    bool requires_seekable_file { false };
+    ErrorOr<NonnullRefPtr<FileSystem>> (*create_with_fd)(OpenFileDescription&, Span<u8 const>) = nullptr;
+    ErrorOr<NonnullRefPtr<FileSystem>> (*create)(Span<u8 const>) = nullptr;
+    ErrorOr<void> (*handle_mount_boolean_flag)(Span<u8>, StringView key) = nullptr;
```

Comment:
It appears the boolean value argument is missing here, or is this function always going to flip the key's value? If so, I don't think that's a good design, since there may be varying defaults, some false, some true.